### ISSUE TITLE
Staging Review finding: Action on input for sort dropdown

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#98157]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

This ticket was part of a merged Staging Review with MHV Medications [List Filter](https://github.com/department-of-veterans-affairs/va.gov-team/issues/88958) and [Prescription Information](https://github.com/department-of-veterans-affairs/va.gov-team/issues/88952)

## Product Information
**Team:** MHV Medications
**Product:** Medications
**Feature:** List Filter & Prescription Information

## Findings details
**VA.gov Experience Standard - issue:** [User experiences an unexpected interaction with a component or pattern.](https://design.va.gov/about/experience-standards/#user-experiences-predictable-interactions-with-components)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility), @erinrwhite (IA)

### Description

On the [/medications/?page=1](https://staging.va.gov/my-health/medications/?page=1) page, the "Show medications in this order" select element initiates a change in content and a change in focus immediately after an input is selected. This came up in [a big Slack discussion about filtering and sorting](https://dsva.slack.com/archives/C8E985R32/p1726630019711259?thread_ts=1726579725.396949&cid=C8E985R32) which I believe you all took into consideration when designing this pattern. I think the decision was that this experience was preferable to having an extra button, and that most users would expect the sort action to occur immediately rather than having to click an extra button to complete the sort. I'm logging this issue for the sake of further discussion, along with a video of the problematic behavior I observe.

When you interact with a `<select>` via mouse, the expected interaction pattern is to click on the element to open the menu of options, and click on an option to choose it. When you interact with a `<select>` via keyboard, there are two possible interaction patterns:
- Tab to the `<select>`, press [enter] to open the menu of options, arrow up or down through the options, and press enter to choose it.
- Tab to the `<select>`, and arrow up or down through the options to choose one.

The important distinction here is that when you open the menu and arrow through the options, the input value of the `<select>` doesn't change until you press [enter]. But when you just arrow through the options without opening the menu, the value of the `<select>` changes.

For a sighted keyboard user, there's an advantage to opening the menu before arrowing through the options, since you can see all of your options at a glance. But for screen reader users who don't get that visual benefit, arrowing up and down without opening the menu is fewer keystrokes --- there's no advantage to pressing [enter] two extra times.

But as currently coded, an action occurs as soon as the value of the `<select>` changes --- meaning a user who's arrowing up or down through the `<select>` without first opening the menu will have the page content changed and their focus immediately taken to the results on their first keystroke, before they may have even explored all of the options available to them. This is a potentially jarring experience, and violates [WCAG 3.2.2](https://www.w3.org/WAI/WCAG22/Understanding/on-input.html).

See the video included for an example of this experience.

<details><summary>Link, screenshot or steps to recreate</summary>

https://github.com/user-attachments/assets/7aa9adeb-c55a-4e0d-89bd-8e788cc93c4a

</details>

## Recommended action

In the past, I've pointed teams toward including a "Sort" button that initiates the action, rather than it occuring when the value of the `<select>` changes. That's still my preferred solution.

I'll also propose a potential alternative, which is that the action and change of focus occur `onBlur` when a user tabs out of the `<select>`, rather than when the value of the `<select>` changes. This would allow a keyboard user to arrow up and down as much as they want, and then pressing tab would initiate the sort action and move focus to "Showing X - Y of Z." That idea violates [WCAG 3.2.1](https://www.w3.org/WAI/WCAG22/Understanding/on-focus.html) but might be preferable?

As I said, I'm sharing this for the sake of discussion.

## References

- **Accessibility Defect Severity:** [2: High. Should be fixed in 1-2 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-2)
- **WCAG Success Criteria:** [WCAG 3.2.2 On Input - Level A](https://www.w3.org/TR/WCAG21/#on-input)
- **Modality:** Keyboard
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [x] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by rileyorr** [comment]
For reference, the team recreated this interaction issue and discovered the interaction is working correctly on Mac but incorrectly on Windows at the design system level. We've notified the Governance Team in [this slack thread](https://dsva.slack.com/archives/C059Q28DV99/p1733938683687889?thread_ts=1733245815.394329&cid=C059Q28DV99) for next steps.

**Comment by TannerBird98** [comment]
This ticket is waiting on the platform governance team to fix the component. It was found that this was an issue on Windows only, Mac computers did not have this issue. I believe they will close this ticket when fixed.

**Comment by shiragoodman** [comment]
@TannerBird98 I think you meant to state that the Design System team will be fixing the component. Since the issue is not resolved, I'm reopening this ticket. We acknowledge it is not your teams responsibility to fix.

cc @caw310

**Comment by caw310** [comment]
Provide your estimate of 1, 2, 3, 5, 8 or 13 
@Andrew565 - 
@ataker - 
@harshil1793 -  
@it-harrison -
@jamigibbs -
@micahchiang -
@powellkerry -
@rmessina1010 -
@rsmithadhoc -

**Comment by Andrew565** [comment]
It was decided by the DST that we will add a separate "onBlur" event which teams can listen for alongside the current "onChange" event.
